jetty-http/src/main/java/org/eclipse/jetty/http/GZIPContentDecoder.java

```diff
@@ -417,7 +417,9 @@ public ByteBuffer acquire(int capacity)
      */
     public void release(ByteBuffer buffer)
     {
-        if (_pool!=null && buffer!=BufferUtil.EMPTY_BUFFER)
+        @SuppressWarnings("ReferenceEquality")
+        boolean isBufferEmpty = (buffer!=BufferUtil.EMPTY_BUFFER);
```

Comment:
`isTheEmptyBuffer` is a better name than `isBufferEmpty` and has a significantly different meaning, as there can be empty buffers that are not *the* `EMPTY_BUFFER`